### PR TITLE
246464 consume mock update since endpoint

### PR DESCRIPTION
**Pull request by harshal15305**
This pull request introduces support for retrieving S-100 product updates using the "updatesSince" endpoint from the Sales Catalogue Service. It adds a new pipeline node to handle this operation, updates the service interface and implementation, and provides comprehensive unit tests for the new logic. Additionally, it ensures the new node is integrated into the S-100 assembly pipeline and refines how the filter parameter is handled for updates-since requests.

**S-100 Product Updates Retrieval:**

* Added a new method `GetS100ProductUpdatesSinceAsync` to the `IProductService` interface to retrieve updated S-100 product details since a specified datetime.
* Implemented `GetS100ProductUpdatesSinceAsync` in `DefaultProductService`, including query parameter handling, retry logic, error logging, and mapping of API responses to domain objects.

**Pipeline Integration:**

* Introduced a new pipeline node `GetS100ProductUpdatesSinceNode` that calls the new service method, updates job and build state, and handles error scenarios.
* Registered the new node in the S-100 assembly pipeline to ensure it is executed as part of the workflow.

**Parameter Handling:**

* Updated the assembly pipeline parameters to pass only the `SinceDateTime` as the filter for "updatesSince" requests, simplifying the filter construction.

**Testing:**

* Added comprehensive unit tests for `GetS100ProductUpdatesSinceNode`, covering successful execution, error handling, and edge cases such as invalid configuration.

### CODE REVIEW

File: src/UKHO.ADDS.EFS.Orchestrator/Pipelines/Assembly/Nodes/S100/GetS100ProductUpdatesSinceNode.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        public GetS100ProductUpdatesSinceNode(AssemblyNodeEnvironment nodeEnvironment, IProductService productService, ILogger<GetS100ProductUpdatesSinceNode> logger)
+            : base(nodeEnvironment)
+        {
+            _productService = productService ?? throw new ArgumentNullException(nameof(productService));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        public override Task<bool> ShouldExecuteAsync(IExecutionContext<PipelineContext<S100Build>> context)
+        {
+            return Task.FromResult(context.Subject.Job.JobState == JobState.Created && (context.Subject.Job.RequestType == RequestType.UpdatesSince));
```
[the remark this answers is not in the document]
Comment (harshal15305) [reply src/UKHO.ADDS.EFS.Orchestrator/Pipelines/Assembly/Nodes/S100/GetS100ProductUpdatesSinceNode.cs:27]: Done

File: src/UKHO.ADDS.EFS.Orchestrator/Pipelines/Assembly/Nodes/S100/GetS100ProductUpdatesSinceNode.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            {
+                productEditionList = await _productService.GetS100ProductUpdatesSinceAsync(sinceDateTime, productIdentifier, job, Environment.CancellationToken);
+            }
+            catch (Exception)
+            {
+                await context.Subject.SignalAssemblyError();
+                return NodeResultStatus.Failed;
+            }
+
+            if (productEditionList == null || !productEditionList.HasProducts)
```
[the remark this answers is not in the document]
Comment (harshal15305) [reply src/UKHO.ADDS.EFS.Orchestrator/Pipelines/Assembly/Nodes/S100/GetS100ProductUpdatesSinceNode.cs:52]: Done

File: src/UKHO.ADDS.EFS.Infrastructure/Services/DefaultProductService.cs [hunk 3]
```diff
@@ -148,5 +148,55 @@ public async Task<ProductEditionList> GetProductEditionListAsync(DataStandard da
...
+                        .GetAsync(
+                            requestConfiguration =>
+                            {
+                                if (requestConfiguration.QueryParameters is not null)
+                                {
+                                    var queryParams = requestConfiguration.QueryParameters;
+                                    var getProductIdentifierProp = queryParams.GetType().GetProperty("ProductIdentifier");
+                                    if (getProductIdentifierProp != null && getProductIdentifierProp.CanWrite && productIdentifier != DataStandardProduct.Undefined)
+                                    {
+                                        getProductIdentifierProp.SetValue(queryParams, productIdentifier.AsEnum.ToString());
```
[the remark this answers is not in the document]
Comment (harshal15305) [reply src/UKHO.ADDS.EFS.Infrastructure/Services/DefaultProductService.cs:175]: Done

**Review by Avinash15515 — COMMENTED** [review]
Minor observations

**Review by richardAhz — APPROVED** [review]
Looks ok

**Review by harshal11869 — APPROVED** [review]
Looks Good.
